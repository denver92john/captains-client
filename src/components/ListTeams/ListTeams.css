.no-lists-link {
    color: var(--color1);
}

.list-team-item {
    border-bottom: 1px solid black;
    padding: 7px;
}

.list-team-item:last-of-type {
    border-bottom: none;
}

.list-team-item-link {
    color: var(--color1);
    font-weight: 500;
    font-size: 20px;
}