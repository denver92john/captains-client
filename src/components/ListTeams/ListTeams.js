import React, {Component} from 'react';
import {Link} from 'react-router-dom';

class ListTeams extends Component {
    render() {
        const {items} = this.props;
        return (
            <div className="wrapper list-component">
                {(items.length === 0)
                    ? <p>You haven't created any lists yet. Go to the <Link to={`/lists`}>Lists Page</Link> to create a new list.</p>
                    : null}
                <ul className="named-list">
                    {items.map(item =>
                        <Link to={`/team/list/${item.id}`} key={item.id} className="named-list-item">
                            <li>
                                {item.list_name}
                            </li>
                        </Link>
                    )}
                </ul>
            </div>
        );
    }
}

export default ListTeams;