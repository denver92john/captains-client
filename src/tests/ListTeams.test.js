import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import ListTeams from '../components/ListTeams/ListTeams';

describe('ListTeams component', () => {
    const itemsProp = [
        {
            id: 1,
            list_name: 'Joe',
            user_id: 1
        },
        {
            id: 2,
            list_name: 'Green',
            user_id: 1
        }
    ];

    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <ListTeams items={itemsProp} />
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const tree = renderer
            .create(<BrowserRouter>
                    <ListTeams items={itemsProp} />
                </BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})