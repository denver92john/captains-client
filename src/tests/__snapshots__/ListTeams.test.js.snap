// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListTeams component renders UI as expected 1`] = `
<div
  className="wrapper list-component"
>
  <ul
    className="named-list"
  >
    <li
      className="list-team-item"
    >
      <a
        className="list-team-item-link"
        href="/team/list/1"
        onClick={[Function]}
      >
        Joe
      </a>
    </li>
    <li
      className="list-team-item"
    >
      <a
        className="list-team-item-link"
        href="/team/list/2"
        onClick={[Function]}
      >
        Green
      </a>
    </li>
  </ul>
</div>
`;
